enterprise/server/suggestion/suggestion.go

```diff
@@ -55,40 +55,50 @@ func (s *suggestionService) MultipleProvidersConfigured() bool {
 }
 
 func (s *suggestionService) GetSuggestion(ctx context.Context, req *supb.GetSuggestionRequest) (*supb.GetSuggestionResponse, error) {
-	if req.GetInvocationId() == "" {
-		return nil, status.InvalidArgumentErrorf("GetSuggestionRequest must contain a valid invocation_id")
+	if req.GetInvocationId() == "" && req.GetPrompt() == "" {
+		return nil, status.InvalidArgumentErrorf("GetSuggestionRequest must contain a valid invocation_id and/or prompt")
 	}
 
-	chunkReq := &elpb.GetEventLogChunkRequest{
-		InvocationId: req.GetInvocationId(),
-		MinLines:     minLines,
+	prompt := req.GetQuery()
+	if prompt == "" {
+		prompt = defaultPrompt
 	}
 
-	resp, err := eventlog.GetEventLogChunk(ctx, s.env, chunkReq)
-	if err != nil {
-		log.Errorf("Encountered error getting event log chunk: %s\nRequest: %s", err, chunkReq)
-		return nil, err
-	}
+	if req.GetInvocationId() != "" {
+		chunkReq := &elpb.GetEventLogChunkRequest{
+			InvocationId: req.GetInvocationId(),
+			MinLines:     minLines,
+		}
 
-	errorMessage := string(resp.GetBuffer())
-	components := strings.SplitN(errorMessage, "ERROR:", 2) // Find the first ERROR: line
-	if len(components) > 1 {
-		errorMessage = components[1]
-	}
-	if len(errorMessage) > maxChars {
-		errorMessage = errorMessage[:maxChars] // Truncate to avoid going over api input limit.
+		resp, err := eventlog.GetEventLogChunk(ctx, s.env, chunkReq)
+		if err != nil {
+			log.Errorf("Encountered error getting event log chunk: %s\nRequest: %s", err, chunkReq)
+			return nil, err
+		}
+
+		errorMessage := string(resp.GetBuffer())
+		components := strings.SplitN(errorMessage, "ERROR:", 2) // Find the first ERROR: line
+		if len(components) > 1 {
+			errorMessage = components[1]
+		}
+		if len(errorMessage) > maxChars {
+			errorMessage = errorMessage[:maxChars] // Truncate to avoid going over api input limit.
+		}
+
+		prompt = prompt + " " + errorMessage
```

Comment:
You could put this in a method if you wanted

Reply:
Good call, I thought about it but was lazy. Done.